File: 백준/Silver/2606. 바이러스/바이러스.py
# 백준 2606번 : 바이러스
# 알고리즘 & 자료구조 : BFS를 통해 구현하면 될 것 같음

from collections import deque

def bfs(i,q,n):
    count = 0
    is_check = [False] * (n+1)

    q.append(i) # 1번 컴퓨터 입력
    is_check[i] = True

    while q :
        num = q.popleft()
        if num != 1 : count += 1 # 1을 통해 걸린 컴퓨터 개수를 세는 것이므로 1은 제외

        for x in computer[num]:
            if not is_check[x] : # 방문하지 않았던 노드라면
                q.append(x) # 큐에 추가
                is_check[x] = True # 방문했다고 표시

    return count

n = int(input()) # 컴퓨터의 수 , 1번부터 번호가 매겨짐
pair = int(input())
computer = [[] for _ in range(n+1)]
q = deque()

for i in range(pair): # 연결된 리스트 만들어주기
    a,b = list(map(int, input().split()))
    computer[a].append(b)
    computer[b].append(a)

res = bfs(1,q,n)
print(res)

File: 백준/Silver/2606. 바이러스/바이러스.java
import java.util.*;
import java.io.*;

public class Main {
	static Deque<Integer> deque = new ArrayDeque<>();
	static boolean[] visit;
	static int[][] computer;
	public static void main(String[] args) throws IOException{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
		
		int num = Integer.parseInt(br.readLine());
		int n = Integer.parseInt(br.readLine());
		computer = new int[num+1][num+1];
		visit = new boolean[num+1];
		
		for (int i = 0; i < n; i++) {
			int[] tmp = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
			computer[tmp[0]][tmp[1]] = 1; // 연결된 바이러스 넣기
			computer[tmp[1]][tmp[0]] = 1;
		}
		
		System.out.println(bfs(1));

	}
	private static int bfs(int num) {
		deque.add(num);
		visit[num] = true; // 방문했다고 표시
		int count = 0;
		
		while(!deque.isEmpty()) {
			int idx = deque.poll();
			
			for (int i = 1; i < computer.length; i++) {
				if(computer[idx][i]==1 &&!visit[i]) {
					deque.add(i); // 큐에 넣어주기
					visit[i] = true; // 방문했다고 표시
					count+=1; // 1번에 연결된 컴퓨터 수 증가
				}
			}
		}
		return count;
		
	}

}